File: alg cpp/1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
    public:
        ListNode* deleteDuplicates(ListNode* head) {
            ListNode* current = head;
            while (current && current->next) {
                if (current->val == current->next->val) {
                    // Remove the duplicate node.
                    ListNode* temp = current->next;
                    current->next = current->next->next;
                    delete temp;
                } else {
                    // Move to the next distinct element.
                    current = current->next;
                }
            }
            return head;
        }
    };
    

File: alg cpp/1.java
// Definition for singly-linked list.
public class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { 
        this.val = val; 
    }
    ListNode(int val, ListNode next) { 
        this.val = val; 
        this.next = next; 
    }
}

public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode current = head;
        // Traverse the list until we reach the end
        while (current != null && current.next != null) {
            // If the current value is the same as the next one, skip the duplicate
            if (current.val == current.next.val) {
                current.next = current.next.next;
            } else {
                // Otherwise, move to the next node
                current = current.next;
            }
        }
        return head;
    }
}
